### PR TITLE
add support for system-usernames

### PR DESCRIPTION
**Pull request by cjp256**
LP #1845267

### CODE REVIEW

**Comment by cjp256** [comment]
Added the remaining functionality to implement and schema tests. It's a little ugly due to "system-usernames" not matching the __dict__ model, but that's going to be removed & cleaned up in an upcoming PR.

File: snapcraft/internal/meta/snap.py [hunk 1]
```diff
@@ -267,6 +279,34 @@ def from_dict(cls, snap_dict: Dict[str, Any]) -> "Snap":
...
+                        )
+                    elif isinstance(raw_user, str):
+                        try:
+                            scope = SystemUserScope[raw_user.upper()]
+                        except KeyError:
+                            raise errors.SystemUsernamesValidationError(
+                                name=user_name, message=f"scope {raw_user!r} is invalid"
+                            )
+                        user = SystemUser(name=user_name, scope=scope)
+                    else:
```
Comment (sergiusens) [inline snapcraft/internal/meta/snap.py]: if this is part of passthrough, it probably should be allowed to have other values here.

Comment (cjp256) [reply snapcraft/internal/meta/snap.py]: I'm not sure what you mean?

Comment (sergiusens) [reply snapcraft/internal/meta/snap.py]: no worries, we can cross that bridge later.

**Pull request merged**